data-service/src/test/java/com/amazonaws/blox/dataservice/integration/DataServiceIntegrationTestConfig.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.integration;
+
+import com.amazonaws.blox.dataservice.mapper.ApiModelMapper;
+import com.amazonaws.blox.dataservice.mapper.EnvironmentMapper;
+import org.mapstruct.factory.Mappers;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class DataServiceIntegrationTestConfig {
```

Comment:
You can load most of the data service configuration directly.   I'm not sure if we can directly import the `Application.class` and replace only the `@Bean public AmazonDynamoDB dynamoDBClient()` like below. If we can override it,  then we load almost the same as real runtime. Otherwise, you can load other beans by ComponentScan.
```
  @Bean
  public AmazonDynamoDB dynamoDBClient() {
    amazonDynamoDBLocal = DynamoDBEmbedded.create();
    return amazonDynamoDBLocal.amazonDynamoDB();
  }
```

Reply:
Sure, I can load the other configs from Application.class. The problem is with the dynamoDB local -- the DDBlocal should be created in the @before method and should be shutdown in the @after method. I tried to create the dynamoDBClient here, but I can't shutdown the amazonDynamoDBLocal later. Any suggestions on this?

Reply:
I think the purpose for shutdown() is just to delete all the created tables. If we don't call it at the end, we can either manually delete all tables after each test, or write the test in a way that it can use the content from the previous tests. I can try deleting the table and see if it works fine.

Reply:
I don't think it will delete the tables. I  should stop all the jobs and release resources. It may only have value if you create a DynamoDBEmbedded to a local file as `DynamoDBEmbedded.create(...File...)`, as the shutdown can release the resource on the file. But for in-memory, we should be safe.

Reply:
Tried with overriding the dynamoDBClient bean. The creation was fine, but the next test will complain that the table has already been created. I think it's because the resource only gets released after we're done with all the unit tests (at the end of the process), but not after each unit test. 
I go with deleting the tables after each test and it's fine...

Reply:
Could we use a `@DirtiesContext` to spin up a new context for each test case?

Reply:
Changing into @DirtiesContext also works!